common/scripts/select-packages.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+
+// This script searches for package.json files recursively under a given path
+// excluding packages under a node_modules folder and opens them to determine
+// whether they contain a "sdk-archetype" field with the value of "client".
+//
+// If they do, then the package name is recorded and then after all package.json files
+// have been scanned a single string is output using the Azure Pipelines Commands special
+// syntax which creates a variable for use within a later stage of the pipeine.
+
+console.log(
+  '##vso[task.setvariable variable=GeneratedPackageTargets;]--to "@azure/core-http" --to "@azure/amqp-common" --to "@azure/abort-controller"'
```

Comment:
I'm not sure where this is getting used. Can you help me understand when it gets picked?

Reply:
This leverages the [##vso commands syntax](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md). The STDIO from build scripts is processed by the agent and when it encounters the patterns described in that doc it takes specific actions.

In this case you can use it to propagate state from one step to another _within_ the same job. To propagate these variables out you set the isOutput flag, but to help with scoping the variables you import the propagated variables via the variables statement. Basically allowing you to map state from one [job].[step] to another job.

At the moment propagation between stages isn't possible but its a known ask.

Reply:
I understand the vso setvariable thing I was asking more about what is special for these core packages that requires them to have there own select-packages.js and I couldn't figure out where it is actually consumed.